input/ipc.c

```diff
@@ -140,7 +140,7 @@ static char *json_execute_command(struct mpv_handle *client, void *ta_parent,
 
     async_node = node_map_get(&msg_node, "async");
     if (async_node) {
-        if (async_node->format != MPV_FORMAT_FLAG) {
+        if (async_node->format != MPV_FORMAT_FLAG || async_node->format != MPV_FORMAT_BOOL) {
```

Comment:
Note to self: actually check this later.

Reply:
I'm pretty sure that condition will always evaluate to true.

Reply:
Yup, should have been `&&`. But I don't think this condition needs to be changed at all. It should come from a client so it should still be only `MPV_FORMAT_FLAG`.

Reply:
Now it will always be MPV_FORMAT_FLAG, but isn't the whole point that eventually MPV_FORMAT_FLAG gets removed and all clients use MPV_FORMAT_BOOL instead?

Reply:
Yeah that's the idea. But I don't think it makes any sense for there to be conditions in the code that will never actually be executed does it? Off in the future, the hypothetical "remove MPV_FORMAT_FLAG" commit would have to change all these places anyway.

Unfortunately MPV_FORMAT_BOOL may just have to be reduced to a simple alias. I wanted it to actually store bool types but I'm not sure that can be done.

Reply:
It should be possible to internally only store bool and then convert between them whenever a flag gets requested or set.
I think reading already does this, but writing still needs to be adjusted.
Something like this maybe?
``` diff
diff --git a/player/client.c b/player/client.c
index 9b4c83b861..29f2faf782 100644
--- a/player/client.c
+++ b/player/client.c
@@ -1294,6 +1294,10 @@ static void setproperty_fn(void *arg)
         memcpy(&tmp.u, req->data, type->type->size);
         node = &tmp;
     }
+    if (node->format == MPV_FORMAT_FLAG){
+        node->format = MPV_FORMAT_BOOL;
+        node->u.bool_ = !!node->u.flag
+    }
 
     int err = mp_property_do(req->name, M_PROPERTY_SET_NODE, node, req->mpctx);
 
```